### PR TITLE
feat: update to storybook 7, webpack 5, node 18

### PR DESCRIPTION
**Pull request by gorgeousvlad**

### CODE REVIEW

File: demo/DocsDecorator/DocsDecorator.scss [hunk 1]
```diff
@@ -0,0 +1,15 @@
+$root: '.docs-decorator';
+
+#{$root}#{$root}#{$root}#{$root}#{$root} {
```
Comment (Kyzyl-ool) [inline demo/DocsDecorator/DocsDecorator.scss:3]: what caused it?)

File: .storybook/addons/theme-addon/register.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import {addons, types} from '@storybook/addons';
+import {useGlobals} from '@storybook/api';
```
[the remark this answers is not in the document]
Comment (Kyzyl-ool) [reply .storybook/addons/theme-addon/register.tsx:4]: Sorry, it also exists in @storybook/api

File: .storybook/addons/theme-addon/register.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        render: () => {
+            return <Tool api={api} />;
+        },
+    });
+});
+
+function Tool({api}: {api: API}) {
+    const [{theme}] = useGlobals();
+    React.useEffect(() => {
+        api.setOptions({theme: themes[theme]});
```
Comment (Kyzyl-ool) [inline .storybook/addons/theme-addon/register.tsx:25]: the standard storybook theme switcher did not work?

Comment (Kyzyl-ool) [reply .storybook/addons/theme-addon/register.tsx:25]: you can also change global theme via:
```
const [globals, updateGlobals] = useGlobals();
...
updateGlobals({theme});
...
```

File: demo/DocsDecorator/DocsDecorator.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import './DocsDecorator.scss';
+
+export interface DocsDecoratorProps extends React.PropsWithChildren<DocsContainerProps> {}
+
+const b = block('docs-decorator');
+
+export function DocsDecorator({children, context}: DocsDecoratorProps) {
+    // @ts-expect-error
+    const theme = context.store.globals.globals.theme;
```
Comment (Kyzyl-ool) [inline demo/DocsDecorator/DocsDecorator.tsx:18]: could we use here `useGlobals`?

**Review by NikitaCG — DISMISSED** [review]

File: demo/DocsDecorator/DocsDecorator.scss (hunk 1 above)
Comment (gorgeousvlad) [reply demo/DocsDecorator/DocsDecorator.scss]: honestly I didn't check it, just copied from uikit)
https://github.com/gravity-ui/uikit/blob/4519ea5b3baa3a0e665a8b477e73ef8ea5cb44a3/src/demo/DocsDecorator/DocsDecorator.scss#L10

File: .storybook/addons/theme-addon/register.tsx (hunk 3 above)
Comment (gorgeousvlad) [reply .storybook/addons/theme-addon/register.tsx:25]: same, it is from uikit, just trying to keep gravity-ui projects similar

**Review by NikitaCG — DISMISSED** [review]

**Review by NikitaCG — APPROVED** [review]

**Review by yuberdysheva — APPROVED** [review]

File: .storybook/addons/theme-addon/register.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import {addons, types} from '@storybook/addons';
+import {useGlobals} from '@storybook/api';
+import type {API} from '@storybook/api';
```
Comment (Kyzyl-ool) [inline .storybook/addons/theme-addon/register.tsx:5]: Maybe we should add `@storybook/api@7.1.0` to dependencies?

Comment (Kyzyl-ool) [reply .storybook/addons/theme-addon/register.tsx:5]: Nevermind though, it is up to you)

**Review by Kyzyl-ool — APPROVED** [review]

**Pull request merged**